packages/csharp/PortingAssistant/PortingAssistant.Telemetry/LogWatcher.cs

```diff
@@ -0,0 +1,278 @@
+
+                        PublishLogs(logs, profile, telemetryConfiguration);
+                    }
+                }
+            }
+            else
+            {
+                Log.Information("File Locked");
+            }
+        }
+
+        private void FileSystemWatcher_Deleted
+            (
+            Object sender,
+            FileSystemEventArgs e,
+            Program.TelemetryConfiguration telemetryConfiguration,
+            string userData
+            )
+        {
+            if (!Path.GetExtension(e.FullPath).Equals(".log")) return;
+
+            FileInfo fileInfo = new FileInfo(e.FullPath);
+            var fileName = e.Name;
+
+            const string LastReadTokenFileName = "lastToken.json";
+            var lastReadTokenFile = Path.Combine(userData, telemetryConfiguration.LogsPath, LastReadTokenFileName);
+
+            Dictionary<string, int> fileLineNumberMap;
+
+            // If log file is deleted set Last Read Line Number to 0.
+            if (File.Exists(lastReadTokenFile))
+            {
+                fileLineNumberMap = JsonConvert.DeserializeObject<Dictionary<string, int>>(File.ReadAllText(lastReadTokenFile));
+                if (fileLineNumberMap.ContainsKey(fileName)) fileLineNumberMap[fileName] = 0;
+            }
+        }
+        private static void PublishLogs
+            (
+            ArrayList logs,
+            string profile,
+            Program.TelemetryConfiguration telemetryConfiguration
+            )
+        {
+            Dictionary<string, ArrayList> logTypeMap = new Dictionary<string, ArrayList>();
+
+            foreach (string log in logs)
+            {
+                var jsonObj = JObject.Parse(log);
+                var content = jsonObj["Content"].ToString(Formatting.None);
+                var type = jsonObj["Type"].ToString();
+
+                if (!logTypeMap.ContainsKey(type)) logTypeMap[type] = new ArrayList();
+                logTypeMap[type].Add(content);
+            }
+
+            foreach (KeyValuePair<string, ArrayList> entry in logTypeMap)
+            {
+                PutLogData(entry.Key, JsonConvert.SerializeObject(entry.Value), profile, telemetryConfiguration);
+            }
+        }
+
+        private static async void PutLogData
+            (
+            string logName,
+            string logData,
+            string profile,
+            Program.TelemetryConfiguration telemetryConfiguration
+            )
+        {
+            const string PathTemplate = "/put-log-data";
+
+            var chain = new CredentialProfileStoreChain();
+            AWSCredentials awsCredentials;
+            var profileName = profile;
+            var region = telemetryConfiguration.Region;
+            if (chain.TryGetAWSCredentials(profileName, out awsCredentials))
+            {
+                var signer = new AWS4RequestSigner
+                    (
+                    awsCredentials.GetCredentials().AccessKey,
```

Comment:
I remember this function may read the default aws credential in customer local configuration file. But it may not match the credential customer picked for the tools. Correct me if I am wrong